### ISSUE TITLE
Kompile issues

### ISSUE THREAD

**Issue by cos** [kframework/k-legacy#1718]
How does the `compileRule` method relate to the other transformations applied to rules as part of the `defaultSteps`?

As `Kompile` starts to take shape, we should add more documentation to it. One rule of thumb would be to not accept pull requests when the public methods that have been touched do not have proper documentation.

**Comment by grosu** [comment]
Of course!!!   I said that all along, that all public methods must have documentation.  @dwightguth , I thought that you were enforcing that already.  So yes, everybody please reject pull requests where variables and methods do not have suggestive names, and where the public methods are not documented properly.  If not sure if a name is good enough or a public method is not documented well enough, please have me revise it.
